server/src/main/java/io/crate/execution/engine/aggregation/impl/average/IntervalAverageAggregation.java

```diff
@@ -0,0 +1,283 @@
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.engine.aggregation.impl.average;
+
+import static io.crate.execution.engine.aggregation.impl.average.AverageAggregation.NAMES;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.joda.time.Period;
+
+import io.crate.Streamer;
+import io.crate.breaker.RamAccounting;
+import io.crate.data.Input;
+import io.crate.execution.engine.aggregation.AggregationFunction;
+import io.crate.execution.engine.aggregation.impl.AggregationImplModule;
+import io.crate.memory.MemoryManager;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import io.crate.types.FixedWidthType;
+import io.crate.types.IntervalType;
+
+public class IntervalAverageAggregation extends AggregationFunction<IntervalAverageAggregation.IntervalAverageState, Period> {
+
+    static {
+        DataTypes.register(IntervalAverageStateType.ID, in -> IntervalAverageStateType.INSTANCE);
+    }
+
+    /**
+     * register as "avg" and "mean"
+     */
+    public static void register(AggregationImplModule mod) {
+        for (var functionName : NAMES) {
+            mod.register(
+                Signature.aggregate(
+                    functionName,
+                    DataTypes.INTERVAL.getTypeSignature(),
+                    DataTypes.INTERVAL.getTypeSignature()),
+                IntervalAverageAggregation::new
+            );
+        }
+    }
+
+    public static class IntervalAverageState implements Comparable<IntervalAverageState> {
+
+        private Period sum = new Period();
+        private long count = 0;
+
+        public Period value() {
+            if (count > 0) {
+                return new Period(sum.toStandardDuration().getMillis() / count);
+            } else {
```

Comment:
Just a question: is it aligned with PG?
There can be another ways:
1. this approach but similar to `age`, normalize output like 
`new Period(sum.toStandardDuration().getMillis() / count).normalizedStandard(PeriodType.yearMonthDayTime())`
`PeriodType.yearMonthDayTime()` is aligned with PG default representation but not sure it's relevant here for avg.

2. computing average per unit. - `new Period().days(cureDays/count).hours(curHours./count)` and so on except months and years which shouldn't be  among units anyway as it's checked in `toStandardDuration()` internally via `checkYearsAndMonths` - but I think this is overkill.

To sum up, this approach looks good but I think we need to add the following things to the docs:

1. `toStandardDuration() `internally throws an exception if Period has months/year units - maybe we need to add an example and mention that in docs that it will throw an error. Maybe we could also check same example in PG 

2. add a working example and maybe explicitly mention that avg is NOT avg per units (ie. NOT approach 2)

Reply:
Thanks a lot for diving into this @BaurzhanSakhariev !
- PostgreSQL uses 1st approach (not an avg per unit but a total average in millis)
- I will use our custom `toStandardDuration()` which works for months and years with 30 and 365 days respectively.

Reply:
postgres doesn't normalize to months/years but only up to days, for avg();
I replicated the integ test in psql:
```
matriv=> SELECT avg('2023-05-04 22:33:11.123'::timestamp - ts) FROM tbl;
          avg           
------------------------
 125 days 25:42:24.4535
(1 row)

```